### PR TITLE
Update webpack for blocks and other domains

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR creates a separate endpoint for blocks and also updates webpack to latest CRA to solve the outstanding issue mentioned in #2854

### CODE REVIEW

File: config/env.js [hunk 1]
```diff
@@ -46,48 +44,56 @@ dotenvFiles.forEach(dotenvFile => {
 // It works similar to `NODE_PATH` in Node itself:
 // https://nodejs.org/api/modules.html#modules_loading_from_the_global_folders
 // Note that unlike in Node, only *relative* paths from `NODE_PATH` are honored.
-// Otherwise, we risk importing Node.js core modules into an app instead of Webpack shims.
```
Comment (nicolad) [inline config/env.js:49]: ha, seems like even CRA team was writing it wrong :)

Comment (nicolad) [reply config/env.js:49]: image

File: config/env.js [hunk 2]
```diff
@@ -9,35 +9,33 @@ delete require.cache[require.resolve('./paths')];
 
 const NODE_ENV = process.env.NODE_ENV;
 if (!NODE_ENV) {
```
Comment (nicolad) [inline config/env.js:11]: because of prettier it's a bit hard to review this file

File: config/getHttpsConfig.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+const chalk = require('react-dev-utils/chalk');
+const paths = require('./paths');
+
+// Ensure the certificate and key provided are valid and if not
+// throw an easy to debug error
+function validateKeyAndCerts({ cert, key, keyFile, crtFile }) {
```
Comment (nicolad) [inline config/getHttpsConfig.js:11]: cool, source: https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/config/getHttpsConfig.js

File: config/paths.js [hunk 4]
```diff
@@ -1,47 +1,29 @@
...
 const fs = require('fs');
-const url = require('url');
 
-const ASSETS = 'assets';
 const ASSETS_FOLDER = 'assets';
 const DIST_FOLDER = 'dist';
 const SOURCE_FOLDER = 'src';
 const ASSETS_DIST_FOLDER = `${ASSETS_FOLDER}/${DIST_FOLDER}`;
 const ASSETS_SOURCE_FOLDER = `${ASSETS_FOLDER}/${SOURCE_FOLDER}`;
+const getPublicUrlOrPath = require('react-dev-utils/getPublicUrlOrPath');
```
Comment (nicolad) [inline config/paths.js:9]: looks cool: https://github.com/facebook/create-react-app/blob/master/packages/react-dev-utils/getPublicUrlOrPath.js

File: config/paths.js [hunk 5]
```diff
@@ -1,47 +1,29 @@
...
-	if (hasSlash && !needsSlash) {
-		return inputPath.substr(0, inputPath.length - 1);
-	} else if (!hasSlash && needsSlash) {
-		return `${inputPath}/`;
-	} else {
-		return inputPath;
-	}
-}
-
-const getPublicUrl = (appPackageJson) => envPublicUrl || require(appPackageJson).homepage;
```
Comment (nicolad) [inline config/paths.js:32]: yeah, also noticed that some CRA stuff was adding some useless noise

File: config/webpack.config.js [hunk 6]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
Comment (nicolad) [inline config/webpack.config.js:1]: why not strict ?

File: config/webpack.config.js [hunk 7]
```diff
@@ -14,15 +12,12 @@ const OptimizeCSSAssetsPlugin = require('optimize-css-assets-webpack-plugin');
 const safePostCssParser = require('postcss-safe-parser');
 const ManifestPlugin = require('webpack-manifest-plugin');
 const InterpolateHtmlPlugin = require('react-dev-utils/InterpolateHtmlPlugin');
-const WorkboxWebpackPlugin = require('workbox-webpack-plugin');
```
Comment (nicolad) [inline config/webpack.config.js:17]: this has been removed, but is still in package.json, do we need this?

Comment (nicolad) [reply config/webpack.config.js:17]: image

File: config/webpack.config.js (hunk 6 above)
Comment (manzoorwanijk) [reply config/webpack.config.js:1]: I didn't remove it

File: config/webpack.config.js (hunk 7 above)
Comment (manzoorwanijk) [reply config/webpack.config.js:17]: May be for offline service workers, but lets remove it for now.

File: config/webpack.config.js [hunk 8]
```diff
@@ -14,15 +12,12 @@ const OptimizeCSSAssetsPlugin = require('optimize-css-assets-webpack-plugin');
...
 const WatchMissingNodeModulesPlugin = require('react-dev-utils/WatchMissingNodeModulesPlugin');
-const ModuleScopePlugin = require('react-dev-utils/ModuleScopePlugin');
 const getCSSModuleLocalIdent = require('react-dev-utils/getCSSModuleLocalIdent');
 const ModuleNotFoundPlugin = require('react-dev-utils/ModuleNotFoundPlugin');
 const ForkTsCheckerWebpackPlugin = require('react-dev-utils/ForkTsCheckerWebpackPlugin');
 const typescriptFormatter = require('react-dev-utils/typescriptFormatter');
 const postcssNormalize = require('postcss-normalize');
-const miniExtract = require('mini-css-extract-plugin');
```
Comment (nicolad) [inline config/webpack.config.js:25]: oops, it was a duplicate

File: config/webpack.config.js (hunk 6 above)
Comment (nicolad) [reply config/webpack.config.js:1]: someone hacked your computer

File: config/webpack.config.js [hunk 9]
```diff
@@ -399,7 +392,6 @@ module.exports = function (webpackEnv) {
 								configFile: false,
 								compact: false,
 								presets: [[require.resolve('babel-preset-react-app/dependencies'), { helpers: true }]],
-								plugins: ['@babel/plugin-proposal-class-properties'],
```
Comment (nicolad) [inline config/webpack.config.js:402]: :+1:

File: config/webpack.config.js (hunk 7 above)
Comment (manzoorwanijk) [reply config/webpack.config.js:17]: Done

**Pull request merged**